### PR TITLE
Added LinkedInAuthentication ApplyRedirect

### PR DESCRIPTION
**Pull request by MartinPaulLippert**
Added ApplyRedirect to LinkedInAuthentication Provider
Changed LinkedInAuthenticationHandler to use ApplyRedirect
Added LinkedInApplyRedirectContext

Changes allow a developer to override the authorization endpoint

### CODE REVIEW

File: src/Owin.Security.Providers.LinkedIn/Provider/LinkedInApplyRedirectContext.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+﻿using Microsoft.Owin;
+using Microsoft.Owin.Security;
+using Microsoft.Owin.Security.Provider;
+
+
+namespace Owin.Security.Providers.LinkedIn
+{
+    /// <summary>
+    /// Context passed when a Challenge causes a redirect to authorize endpoint in the Facebook middleware
```
Comment (TerribleDev) [inline src/Owin.Security.Providers.LinkedIn/Provider/LinkedInApplyRedirectContext.cs]: these comments do not appear to be correct... facebook middleware?

File: src/Owin.Security.Providers.LinkedIn/Provider/ILinkedInAuthenticationProvider.cs [hunk 2]
```diff
@@ -20,5 +20,11 @@ public interface ILinkedInAuthenticationProvider
         /// <param name="context"></param>
         /// <returns>A <see cref="Task"/> representing the completed operation.</returns>
         Task ReturnEndpoint(LinkedInReturnEndpointContext context);
+
+        /// <summary>
+        /// Called when a Challenge causes a redirect to authorize endpoint in the Facebook middleware
```
Comment (TerribleDev) [inline src/Owin.Security.Providers.LinkedIn/Provider/ILinkedInAuthenticationProvider.cs]: are you sure this comment is accurate?

**Comment by TerribleDev** [comment]
deployed in 2.11.0

**Comment by MartinPaulLippert** [comment]
Thank you

**Pull request merged**